cross-cluster/openshift/x-cluster-migrate

```diff
@@ -0,0 +1,392 @@
+
+  # rename files to correct profile
+  SRC_PARAM_NAME="./templates/source-pvc-migrator/source-pvc-migrator-deploy.local.param"
+  DST_PARAM_NAME="./templates/target-pvc-migrator/target-pvc-migrator-deploy.local.param"
+
+  
+
+  updateMigratorDeplymentParams ${SRC_PVC} ${DST_PVC} ${REMOTE_CLUSTER} ${REMOTE_PROJECT} ${SRC_CLUSTER} ${DST_CLUSTER} "${SRC_PARAM_NAME} ${DST_PARAM_NAME}"
+}
+
+deployToCluster(){
+  _context=${1}
+  _cluster=${2}
+  _profile=${3}
+
+  # make sure we're using the destination context
+  printStatusMsg "Switching to ${_profile} cluster"
+  oc config use-context ${_context}
+
+  if [ ${_profile} = "source" ]; then
+    oc apply -n ${_cluster} -f ./tmp.secret
+    oc label secret x-cluster-test base=xc-pvc-migrator -n ${_cluster}
+  fi
+  
+  # regenerate settings for deployment
+  unset SETTINGS_LOADED
+  genDepls.sh -c "${_profile}"-pvc-migrator -n "${_context}"
+
+  printStatusMsg "Monitor the deployment in ${_cluster} console and press ENTER after it completes:"
+  read
+
+}
+
+
+targetDeploy(){
+  _context=${1}
+  _cluster=${2}
+  
+  deployToCluster ${_context} ${_cluster} "target"
+
+  MIG_TOKEN_SECRET=`oc get secret | grep target-pvc-migrator-token | head -1 | awk '{print $1}'`
+
+  # Use token value to generate local temp secret
+  oc create secret generic x-cluster-test --from-literal=token=`oc get secret ${MIG_TOKEN_SECRET} -o json | jq -r .data.token | base64 -d` --dry-run -o json > tmp.secret
+}
+
+sourceDeploy(){
+  _context=${1}
+  _contextDst=${2}
+  _cluster=${3}
+  #swtich to source cluster -------------------
+  printStatusMsg "Switching to destiantion cluster"
+  oc config use-context ${_contextDst}
+
+  #grab the target pod name
+  MIG_POD_TARGET=$(oc get pods | grep target-pvc-migrator | awk '{print $1}')
+
+  deployToCluster ${_context} ${_cluster} "source"
+
+  MIG_POD=$(oc get pods | grep source-pvc-migrator | awk '{print $1}')
+
+  #runInContainer doesn't allow us to specify -n or our env, we have to use this instead
+  oc exec -n ${_cluster} -it "${MIG_POD}" -- bash -c "cd /source; oc rsync --compress=true --kubeconfig=/target/tkube/KUBECONFIG . ${MIG_POD_TARGET}:/target"
+}
+
+migrate(){
+    targetDeploy ${DST_CONTEXT} ${DST_CLUSTER}
+    sourceDeploy ${SRC_CONTEXT} ${DST_CONTEXT} ${SRC_CLUSTER}
+}
+
+cleanEnv(){
+  _profile=${1}
+  if [ ! -z ${_profile} ] && [ "${_profile}" = "target" ]; then
+    _context=${DST_CONTEXT}
+  elif [ ! -z ${_profile} ] && [ "${_profile}" = "source" ]; then
+    _context=${SRC_CONTEXT}
+  else
+    printStatusMsg "Please either specifiy target or source profile"
+    exit
+  fi
+
+  printStatusMsg "Cleaning secrets, deployments, builds, etc"
+
+  # make sure we're using the destination context
+  printStatusMsg "Switching to destination cluster"
+  oc config use-context ${_context}
+
+  oc delete -l base=xc-pvc-migrator deployment,sa,rolebinding,secret
+  oc delete -l app=pvc-migrator build,buildconfig,is
+  oc delete -l app=pvc-migrator-nsp nsp
```

Comment:
We should probably be namespacing these commands with `-n something`, I suspect using the context does not limit actions to a specific namespace - if the user has access to more than one on the same cluster.

Reply:
Agreed!

Reply:
added namespace